employees/views.py

```diff
@@ -334,6 +327,48 @@ def post(  # pylint: disable=unused-argument
         return self.redirect_to_another_month(request)
 
 
+@method_decorator(login_required, name="dispatch")
+@method_decorator(check_permissions(allowed_user_types=[CustomUser.UserType.ADMIN.name]), name="dispatch")
+class AuthorReportView(BaseAuthorReportView):
+    url_name = "author-report-list"
```

Comment:
You can fix this as well in `MonthNavigationMixin`, to make it use:
```
from django.urls import resolve
current_url = resolve(request.path_info).url_name
```

This way there is no need to hardcode url name in each view using it.

Reply:
I think I have done it, but not sure because after my change I have do repair many tests from `test_unit_custom_mixins.py`

Reply:
Yes, I see you did :)